### ISSUE TITLE
gdalwarp copies colors regardless of -setci flag

### ISSUE THREAD

**Issue by mxkpp** [OSGeo/gdal#5782]
## Expected behavior and actual behavior.

From [official gdalwarp docs](https://gdal.org/programs/gdalwarp.html), `-setci` causes color tables to be copied from the source to the target.

IMO this implies that omission of `-setci` should do the opposite -- I expected the output raster to have undefined colors.  However, the output raster seems to inherit colors regardless.

## Steps to reproduce the problem.

1. Obtain a raster with color tables already defined -- here: [NLCD_2019_Land_Cover.zip](https://github.com/OSGeo/gdal/files/8764376/NLCD_2019_Land_Cover.zip).
2. Run `gdalwarp` CLI executable with `-overwrite` to create a new raster. Do not specify `-setci` in the call.
3. Load the output raster in QGIS and see that the colors were transferred.

I also tried with `-nomd`.

## Operating system

Windows 10 Enterprise

## GDAL version and provenance

OSGeo4W shell 3.4.2 from https://trac.osgeo.org/osgeo4w/

**Comment by jratike80** [comment]
Color interpretation means a different thing. It defines the meaning of the image band, see a list of possible values https://github.com/OSGeo/gdal/blob/b3e56090fd0890a265e1351100307b597d187056/gcore/gdal.h#L203.

I am not sure if it is possible to drop the color table with existing GDAL utilities. One workaround is to use VRT as an interim output
`gdal_translate NLCD_2019_Land_Cover.tif NLCD_2019_Land_Cover.vrt`

Open the vrt file with text editor, remove the ColorTable block and edit the colorinterpretation into `<ColorInterp>Gray</ColorInterp>`.

Now the image opens without palette into QGIS. If you wish you can save vrt into a new tif with gdal_translate.

I will close the ticket because you did not find a bug. Send further questions about how to use GDAL to the gdal-dev mailing list or into gis.stackexchange that also serves usually pretty well.

**Comment by mxkpp** [comment]
@jratike80, are you sure about that definition for "color interpretation"?

Consider the `-colorinterp` option of [`gdal_translate`](https://gdal.org/programs/gdal_translate.html), described as: "Override the __color interpretation__ of all specified bands.".

As expected, when you pass `-colorinterp None` into `gdal_translate`, the colors are dropped from the output.

I think at the very least the docs should clarify this semantic descrepancy.

**Comment by jratike80** [comment]
I think that perhaps two different kind of colorinterpretations are mixed here. First there is the GDAL ColorInterpretation model that it rather large 

`None` does not seem to be a supported value in gdal_translate but I had a try with `gray` and `undefined` and they both dropped the color table when I converted the NLCD... image into a new TIFF file. I did not know that it would happen and that is a good trick to know, thanks for that.

However, I think that dropping the color table is not totally intentional but it comes from the TIFF specification https://www.itu.int/itudoc/itu-t/com16/tiff-fx/docs/tiff6.pdf. If a TIFF image has a color table then the PhotometricInterpretation must be 3=palette color, and if PhotometricInterpretation is something else (0, 1, 2, or 4) then a color table must not exist. Therefore when gdal_translate is forced to change the color interpretation of the paletted source TIFF into undefined or gray it must drop the color table or otherwise the resulting TIFF would be invalid. But while that happens with TIFF output it does not necessarily happen with other formats that do not have the same limitations than TIFF. Change the outputformat into VRT and GDAL is happy to write a gray band that contains color table (I do not know it is makes sense or not):
```
 <Description>Layer_1</Description>
    <ColorInterp>Gray</ColorInterp>
    <ColorTable>
      <Entry c1="0" c2="0" c3="0" c4="255" />
      <Entry c1="0" c2="0" c3="0" c4="255" />
```

GDAL has a much larger set of ColorInterpretations than the four alternatives in TIFF and for GDAL they all are per band:
```
/*! Types of color interpretation for raster bands. */
typedef enum
{
    /*! Undefined */                                      GCI_Undefined=0,
    /*! Greyscale */                                      GCI_GrayIndex=1,
    /*! Paletted (see associated color table) */          GCI_PaletteIndex=2,
    /*! Red band of RGBA image */                         GCI_RedBand=3,
    /*! Green band of RGBA image */                       GCI_GreenBand=4,
    /*! Blue band of RGBA image */                        GCI_BlueBand=5,
    /*! Alpha (0=transparent, 255=opaque) */              GCI_AlphaBand=6,
    /*! Hue band of HLS image */                          GCI_HueBand=7,
    /*! Saturation band of HLS image */                   GCI_SaturationBand=8,
    /*! Lightness band of HLS image */                    GCI_LightnessBand=9,
    /*! Cyan band of CMYK image */                        GCI_CyanBand=10,
    /*! Magenta band of CMYK image */                     GCI_MagentaBand=11,
    /*! Yellow band of CMYK image */                      GCI_YellowBand=12,
    /*! Black band of CMYK image */                       GCI_BlackBand=13,
    /*! Y Luminance */                                    GCI_YCbCr_YBand=14,
    /*! Cb Chroma */                                      GCI_YCbCr_CbBand=15,
    /*! Cr Chroma */                                      GCI_YCbCr_CrBand=16,
    /*! Max current value (equals to GCI_YCbCr_CrBand currently) */ GCI_Max=16
} GDALColorInterp;

const char CPL_DLL *GDALGetColorInterpretationName( GDALColorInterp );
GDALColorInterp CPL_DLL GDALGetColorInterpretationByName( const char *pszName );

/*! Types of color interpretations for a GDALColorTable. */
typedef enum
{
  /*! Grayscale (in GDALColorEntry.c1) */                      GPI_Gray=0,
  /*! Red, Green, Blue and Alpha in (in c1, c2, c3 and c4) */  GPI_RGB=1,
  /*! Cyan, Magenta, Yellow and Black (in c1, c2, c3 and c4)*/ GPI_CMYK=2,
  /*! Hue, Lightness and Saturation (in c1, c2, and c3) */     GPI_HLS=3
} GDALPaletteInterp;
```

I do not now what issue the `-setci` option fixed when it was added into GDAL version 1.10 https://svn.osgeo.org/gdal/branches/1.10/gdal/NEWS. I have a feeling that it could have something to do with multiband images or if the fourth band is written as "undefined" or "alpha". I am sure that gdalwarp has never dropped color tables by default like you seem to suggest. The documentation of the -setci option could certainly be improved once somebody tells why the option exists. And it would be nice to have a simple utility for dropping the color table and changing the image interpretation of TIFF images by the same. Maybe there are also other formats which have same kind of connection between those two.

**Comment by mxkpp** [comment]
@jratike80, the flag that works in gdal_translate is `-colorinterp undefined`.  I've corrected my previous comment where I suggested `-colorinterp None`.  Too much Python lately.

If `-setci` in `gdalwarp` has any effect, it's extremely subtle and clarity would help. Since the colors are transferred whether this flag is used or not (inconsistent with `gdal_translate`).

Thanks for your time.
